src/map.ts

```diff
@@ -0,0 +1,10 @@
+// We have an ES6 Map available, return the native instance
+
+let exportedMap: MapConstructor;
+if (typeof global.Map !== 'undefined') {
+  exportedMap = global.Map;
+} else {
+  exportedMap = require('core-js/es/map');
```

Comment:
hm shouldn't this otherwise be polyfilled by TypeScript for us?

Reply:
Typescript doesn't do polyfills, that's a babel thing which is strange since we use babel to shim Buffer in browsers, we could have used babel to polyfill Map, that might still be a possible solution here instead of manually importing it (I loathe the directly added dependency). Although on the other hand perhaps we shouldn't be doing this logic of if Map exists on global and instead export the polyfill regardless since the point here is to use it on platforms where Map is not supported.

Reply:
Not sure if this is covered in the bundling but browser would need `window.Map`.